src/App.tsx

```diff
@@ -15,49 +18,96 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  length = 'length',
+  alphabet = 'alphabet',
+}
+
+function getPreparedGoods(
+  goodsList: string[],
+  sortField: string,
+  isReversed: boolean,
+): string[] {
+  const preparedGoods: string[] = [...goodsList];
+
+  if (sortField) {
+    preparedGoods.sort((firstGood: string, secondGood: string) => {
+      switch (sortField) {
+        case SortType.length:
+          return firstGood.length - secondGood.length;
+
+        case SortType.alphabet:
+          return firstGood.localeCompare(secondGood);
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return isReversed
+    ? preparedGoods.reverse()
+    : preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = React.useState('');
+  const [isReversed, setIsReversed] = React.useState(false);
+
+  const visibleGoods = getPreparedGoods(
+    goodsFromServer,
+    sortField,
+    isReversed,
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button is-info', {
+            'is-light': sortField !== SortType.alphabet,
+          })}
+          onClick={() => setSortField(SortType.alphabet)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn('button is-success', {
+            'is-light': sortField !== SortType.length,
+          })}
+          onClick={() => setSortField(SortType.length)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn('button is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={() => setIsReversed(!isReversed)}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(sortField !== '' || isReversed) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setIsReversed(false);
+              setSortField('');
+            }}
```

Comment:
Move to separate function  (transfer the function to a variable, and here only write the name of the function)